src/main/java/org/opensrp/repository/postgres/SettingRepositoryImpl.java

```diff
@@ -220,26 +230,35 @@ private SettingConfiguration convert(org.opensrp.domain.postgres.Settings settin
 	}
 	
 	private SettingsMetadata createMetadata(SettingConfiguration entity, Long id) {
-		SettingsMetadata metadata = new SettingsMetadata();
-		metadata.setSettingsId(id);
-		metadata.setDocumentId(entity.getId());
-		metadata.setIdentifier(entity.getIdentifier());
-		metadata.setServerVersion(entity.getServerVersion());
-		return metadata;
+		
+		try {
+			
+			SettingsMetadata metadata = new SettingsMetadata();
+			metadata.setSettingsId(id);
+			metadata.setDocumentId(entity.getDocumentId() != null ? entity.getDocumentId() : UUID.randomUUID().toString());
+			metadata.setIdentifier(entity.getIdentifier());
+			metadata.setProviderId(entity.getProviderId());
+			metadata.setLocationId(entity.getLocationId());
+			metadata.setTeam(entity.getTeam());
+			metadata.setTeamId(entity.getTeamId());
+			
+			return metadata;
+		}
+		catch (Exception e) {
+			logger.error(e.getMessage(), e);
+			return null;
+		}
 	}
 	
 	@Override
 	public SettingsMetadata saveSetting(SettingConfiguration entity, SettingsMetadata metadata) {
-		if (entity == null || entity.getIdentifier() == null) {
+		if (entity == null || entity.getSettings() == null || entity.getIdentifier() == null) {
 			return null;
 		}
 		
-		if (entity.getId() == null) {
-			entity.setId(UUID.randomUUID().toString());
-		}
 		setRevision(entity);
 		
-		Settings settings = convert(entity, Long.valueOf(entity.getId()));
+		Settings settings = convert(entity, entity.getId() != null ? Long.valueOf(entity.getId()) : null);
```

Comment:
This may bring problems because the `id` is the documentId and not the primary key on table. For that method only the primary key should be passed. 

You can call `org.opensrp.repository.postgres.mapper.custom.CustomSettingMetadataMapper.selectByDocumentId(String)` to get the primaryKey